surfactant/filetypeid/id_extension.py

```diff
@@ -26,6 +26,8 @@ def identify_file_type(filepath: str) -> Optional[str]:
         ".htm": "HTML",
         ".php": "PHP",
         ".bat": "BATCH",
+        ".pl": "PERL",
+        ".pm": "PERL_MODULE",
```

Comment:
If we start trying to parse perl scripts to try to extract some info on what they're doing, is there any difference in how we'd want to treat .pl files vs .pm files?

Reply:
Pm files don't start with a #! which is probably an important identifier. They start with `package <name>;` and then in the .pl file that uses the .pm, there is a top-level `use <pm-name>; `line to import it. Then the .pl file that uses the .pm will call subroutines (functions) from it like a namespace in C++. .pm files can also use other .pm files so there is a dependency hierarchy there. Other than the fact that .pm's are always dependent on .pl's the syntax is identical. From what I understand, you would need to understand the dependency of .pm's on a .pl file if you wanted to recover the semantic information for the subroutines.